### PR TITLE
Show active users modal 

### LINKED ISSUE #2670: View for "Users With Active Tasks"
Build a front-end view for `api/projects/<name>/usersWithOpenTasks` which is available from the projects view.

- [x] add a button to the projects view
- [x] show a modal that displays all users with active tasks of the selected project
- [x] add a button next to each user that if clicked, transfers all its tasks to a selected user

https://discuss.webknossos.org/t/show-active-users/561

### LINKED ISSUE #2862: Add API requests in the backend to get and to transfer all active tasks of a given project
In order to provide the feature of issue [#2670](https://github.com/scalableminds/webknossos/issues/2670) the backend needs to provide two new API requests:

- [x] a API request that provides all users with their number of active annotation tasks as a response (only accessible if the current user is an administrator) 
:arrow_right: maybe override` /api/projects/$name<[^/]+>/usersWithOpenTasks`
- [x] a API request that makes it possible to transfer all active annotation tasks to a single user (also only accessible for administrators)

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
### Mailable description of changes:
 - Admins now can see and transfer all active tasks of a project to a single user

### URL of deployed dev instance (used for testing):
- 

### Steps to test:
- go to the admin -> projects
- use the show active users button on the right side to open the modal
- check the table if its showing all users with their active tasks
- select a user and then transfer the active tasks to the user (a toast should show up to confirm the transfer)
- then check the tasks-tab from the dashboard to confirm the transfer

### Issues:
- fixes #2670
- fixes #2862

------
- [x] Ready for review

### CODE REVIEW

**Comment by jstriebel** [comment]
@MichaelBuessemeyer: Please add the change notes from this PR to the [changelog](https://github.com/scalableminds/webknossos/blob/master/CHANGELOG.md#unreleased)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,62 +1,61 @@
...
-  - Added permission for team managers to create explorational tracings on datasets without allowed teams. [#2758](https://github.com/scalableminds/webknossos/pull/2758)
-  - Added higher-resolution images for dataset gallery thumbnails. [#2745](https://github.com/scalableminds/webknossos/pull/2745)
-  - Added permission for admins to get tasks from all projects in their organization. [#2728](https://github.com/scalableminds/webknossos/pull/2728)
-  - Added the shortcut to copy the currently hovered cell id (CTRL + I) to non-volume-tracings, too. [#2726](https://github.com/scalableminds/webknossos/pull/2726)
-  - Added permission for team managers to refresh datasets. [#2688](https://github.com/scalableminds/webknossos/pull/2688)
-  - Added backend-unit-test setup and a first test for NML validation. [#2829](https://github.com/scalableminds/webknossos/pull/2829)
-
+* Added a button to the administration/project-tab, that displays a modal with all users with active annotations of the selected project. All these annotations can be transfered to a single user [#2670](https://github.com/scalableminds/webknossos/issues/2670)
```
Comment (daniel-wer) [inline CHANGELOG.md]: @MichaelBuessemeyer You accidently re-formatted the whole CHANGELOG.md file (or better, the prettier sublime plugin did). Please revert the changes in this file, add the following to your JsPrettier sublime user config: `"auto_format_on_save_excludes": ["*/node_modules/*","*.md"]` (so .md files are no longer "prettified", and then re-add your changelog entry :)

File: app/assets/javascripts/admin/project/transfer_all_tasks_modal.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  }
+
+  handleSelectChange = (userId: string) => {
+    const selectedUser = this.state.users.find(user => user.id === userId);
+    this.setState({ selectedUser });
+  };
+
+  renderFormContent() {
+    return (
+      <Select
```
Comment (philippotto) [inline app/assets/javascripts/admin/project/transfer_all_tasks_modal.js]: You copied that code from the transfer_task_modal, right? If you feel like it, you can share this code by creating a new "<UserSelectionComponent />" component. The approach would be very similar to the `TeamSelectionComponent` which abstracts the same thing for teams. If you think it's to much for this PR, please add a comment above this line here (e.g., "Todo: create a <UserSelectionComponent /> component which is used here and in transfer_task_modal").

**Comment by philippotto** [comment]
Not sure how I managed to do this, but I think something is not right with the back-end part. @fm3 Can you have a look?

I created a second user and a second project B. For both projects I created some tasks, and requested them for both users. Afterwards, I transferred the tasks. For the task of project B, the tasks route does give me an annotation, but the `task` attribute is null?

image

This leads to an error message (front-end says: "Annotation 5b6d70a27d0000d100b4f2bf has no task assigned. Please inform your admin.") and none of the tasks are rendered in the task lists because of that.

When I transfer all the tasks back to scmboy, I don't get the error anymore. Only the one task of project B is causing this :thinking:

**Comment by fm3** [comment]
@philippotto Good catch! This happens when the user has no read access to the annotation’s dataset. I just pushed a commit that forbids this transfer entirely.
Do we expect to merge this today? Because part of that bug exists in master, so I would create a separate PR to fix it faster.

**Comment by philippotto** [comment]
Just tested it and there is a comprehensible error message :+1: It might be useful to say "transfer all tasks which whose datasets are accessible for the user", but I'd wait until this feature is actually requested from the lab.
